### PR TITLE
Introduce --os-internal-endpoint flag

### LINKED ISSUE #142: svfs uses public storage-url even when you've specified private one
### Context
- svfs version : 0.9.1
- storage provider : Rackspace Swift
- product : Openstack Mitaka

### Steps to reproduce this issue :

1. mount storage using these settings: `svfs mount --device swift --mountpoint /home/user/SWIFT --default-uid=500 --default-gid=500 --os-auth-url=http://%PRIVATE_IP%:5000/v2.0 --os-username=%USER% --os-password=%PASSWORD% --os-tenant-name=swift --os-container-name=container --os-storage-url=http://%private_ip%:8080/v1/AUTH_%ID% --debug`
2. Run it for couple of hours with high load.
3. Suddenly it switches to *https* protocol and *public IP*:

```
time="2017-07-19T19:36:15Z" level=debug msg="-> [ID=0x4edefd] Read 131072" source=fuse 
time="2017-07-19T19:36:15Z" level=debug msg="-> [ID=0x4edef4] Open error=EIO: Get https://%PUBLIC_IP%:8080/v1/AUTH_%ID%/container/file1: x509: cannot validate certificate for %PUBLIC_IP% because it doesn't contain any IP SANs" source=fuse 
time="2017-07-19T19:36:15Z" level=debug msg="-> [ID=0x4edefc] Open error=EIO: Get https://%PUBLIC_IP%:8080/v1/AUTH_%ID%/container/file2: x509: cannot validate certificate for %PUBLIC_IP% because it doesn't contain any IP SANs" source=fuse 
time="2017-07-19T19:36:15Z" level=debug msg="<- Read [ID=0x4edefe Node=0x46bd Uid=505 Gid=506 Pid=20316] 0x2c 110592 @0x5c000 dir=false fl=0 lock=0 ffl=OpenReadOnly" source=fuse
```

### Results you expected :

svfs should always use provided private os-storage-url parameter

### Results you observed :

svfs suddenly switches to public os-storage-url

P.S. I didn't find logs which tell why svfs switched to public https. I just have info that previous operation was successful, then next operation failed because of the certificate validation.

### PR DESCRIPTION
**Pull request by kayrus**
### What I did

Introduced --os-internal-endpoint flag handler

### How I did it

Carefully

### How to verify it

Mount swift storage with the `internal_endpoint` flag

### Description for changelog

Introduced --os-internal-endpoint flag handler. Suitable for private insecure endpoints.

Resolves #142

### CODE REVIEW

File: cmd/mount.go [hunk 1]
```diff
@@ -217,6 +218,7 @@ func setFlags() {
 	viper.BindPFlag("os_region_name", mountCmd.PersistentFlags().Lookup("os-region-name"))
 	viper.BindPFlag("os_auth_token", mountCmd.PersistentFlags().Lookup("os-auth-token"))
 	viper.BindPFlag("os_storage_url", mountCmd.PersistentFlags().Lookup("os-storage-url"))
+	viper.BindPFlag("os_internal_endpoint", mountCmd.PersistentFlags().Lookup("os-internal-endpoint"))
```
Comment (xlucas) [inline cmd/mount.go]: This is not necessary since viper doesn't have to handle this option.

Comment (kayrus) [inline cmd/mount.go]: Was not aware about this. Fixed.

File: cmd/mount.go [hunk 2]
```diff
@@ -287,6 +288,7 @@ func useConfiguration() {
 
 	svfs.SwiftConnection.AuthToken = viper.GetString("os_auth_token")
 	svfs.SwiftConnection.StorageUrl = viper.GetString("os_storage_url")
+	svfs.SwiftConnection.Internal = viper.GetBool("os_internal_endpoint")
```
Comment (xlucas) [inline cmd/mount.go]: Same thing here, remove unnecessary viper call since the option should only handled by cobra.

Comment (kayrus) [inline cmd/mount.go]: done

**Pull request merged**